glommio/src/executor/mod.rs

```diff
@@ -659,7 +736,7 @@ pub struct LocalExecutorPoolBuilder {
     /// How often to yield to other task queues
     preempt_timer_duration: Duration,
     /// Indicates a policy by which [`LocalExecutor`]s are bound to CPUs.
-    placement: Placement,
+    placement: Option<Placement>,
```

Comment:
Why is this an option? Is there any situation in which we expect this to be `None` ?
It seems to me that especially with the fact that `Placement` has a default, we could just use it straight away

Am I missing something ?

Reply:
It does look a bit confusing at first -- I made this an `Option<Placement>` because of ownership rules.  Specifically, inside `on_all_shards` I call a new `spawn_thread` method, which I created to be able to write unit tests for the `Latch`.  The `spawn_thread` method is called in a loop with a `&self`, but I also need to take ownership of the `placement` before calling `spawn_thread` (if all this is done is one method as before, Rust is smart enough to figure out that I'm only taking the `placement` field once).  My options were:

* use an `Option<Placement>` and call `Option::take` (current approach)
* use `std::mem::replace(&mut self.placement, Placement::Unbounded)` but this felt a little misleading inside the `spawn_thread` method (`placement` is not used there, but who knows what might happen in the future)
* destructure `self` and pass all the required fields into `spawn_thread` as separate argument (feels ugly)
* clone `placement`; this would be ok today, but might not work in the future for e.g. `Placement::Custom(f)` where `f` is some closure supplied by the user